pkg/streamingpromql/testdata/ours/native_histograms.test

```diff
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: AGPL-3.0-only
+# Provenance-includes-location: https://github.com/prometheus/prometheus/tree/main/promql/testdata/native_histograms.test
+# Provenance-includes-license: Apache-2.0
+# Provenance-includes-copyright: The Prometheus Authors
+
+# buckets:[1 2 1] means 1 observation in the 1st bucket, 2 observations in the 2nd and 1 observation in the 3rd (total 4).
+load 5m
+	single_histogram	{{schema:0 sum:5 count:4 buckets:[1 2 1]}}
+
+eval instant at 5m single_histogram
+	{__name__="single_histogram"} {{count:4 sum:5 buckets:[1 2 1]}}
+
+eval range from 0 to 4m step 1m single_histogram
+	{__name__="single_histogram"} {{schema:0 sum:5 count:4 buckets:[1 2 1]}}x4
```

Comment:
[nit] This only tests the lookback behaviour - it would be good to add a second sample to `single_histogram` at 5m to test that we correctly pick up the next point once it's valid.